acra-censor/handlers/blacklist_handler.go

```diff
@@ -350,37 +350,60 @@ func handleValuePattern(queryNodes, patternNodes []sqlparser.SQLNode) bool {
 
 //handle SELECT * FROM table %%WHERE%% pattern
 func handleStarPattern(queryNodes, patternNodes []sqlparser.SQLNode) bool {
-	patternStarDetected := false
 	patternWhereDetected := false
 	queryWhereDetected := false
-	var patternTables sqlparser.TableExprs
-	var queryTables sqlparser.TableExprs
-	for _, patternNode := range patternNodes {
-		if _, ok := patternNode.(*sqlparser.StarExpr); ok {
-			patternStarDetected = true
+	patternNodeOffset := 0
+	queryNodeOffset := 0
+	for index := 1; index < len(patternNodes); index++ {
+		if index+patternNodeOffset >= len(patternNodes) || index+queryNodeOffset >= len(queryNodes) {
+			return false
 		}
-		if patternWhereNode, ok := patternNode.(*sqlparser.Where); ok && patternWhereNode != nil && strings.EqualFold(sqlparser.String(patternWhereNode.Expr), WhereConfigPlaceholderReplacerPart2) {
-			patternWhereDetected = true
+		if reflect.DeepEqual(patternNodes[index+patternNodeOffset], queryNodes[index+queryNodeOffset]) {
+			continue
 		}
-		if patternSelectStatement, ok := patternNode.(*sqlparser.Select); ok && patternSelectStatement != nil {
-			patternTables = patternSelectStatement.From
+		if patternSelectExpr, ok := patternNodes[index+patternNodeOffset].(sqlparser.SelectExprs); ok && starFound(patternSelectExpr) {
+			if _, ok := queryNodes[index+queryNodeOffset].(sqlparser.SelectExprs); ok {
+				i := index
+				for i < len(queryNodes) {
+					if _, ok := queryNodes[i].(sqlparser.TableExprs); ok {
+						break
+					}
+					queryNodeOffset++
+					i++
+				}
+				i = index
+				for i < len(patternNodes) {
+					if _, ok := patternNodes[i].(sqlparser.TableExprs); ok {
+						break
+					}
+					patternNodeOffset++
+					i++
+				}
+				continue
+			}
 		}
-	}
-	if !patternStarDetected {
-		return false
-	}
-	for _, queryNode := range queryNodes {
-		if queryWhereNode, ok := queryNode.(*sqlparser.Where); ok && queryWhereNode != nil {
-			queryWhereDetected = true
+		if patternWhereNode, ok := patternNodes[index+patternNodeOffset].(*sqlparser.Where); ok && patternWhereNode != nil && strings.EqualFold(sqlparser.String(patternWhereNode.Expr), WhereConfigPlaceholderReplacerPart2) {
+			patternWhereDetected = true
 		}
-		if querySelectStatement, ok := queryNode.(*sqlparser.Select); ok && querySelectStatement != nil {
-			queryTables = querySelectStatement.From
+		if queryWhereNode, ok := queryNodes[index+queryNodeOffset].(*sqlparser.Where); ok && queryWhereNode != nil {
+			queryWhereDetected = true
```

Comment:
can we return `true` here and `false` in `else` block and don't check `queryWhereDetected && partternWhereDetected`?

Reply:
what if nodes are not both *sqlparser.Where?